### PR TITLE
add catch for exceptions in sample method

### PR DESCRIPTION
**Pull request by jpbempel**
and log it

### CODE REVIEW

File: dd-java-agent/agent-mlt/src/main/java/com/datadog/mlt/sampler/JMXSampler.java [hunk 1]
```diff
@@ -86,27 +86,31 @@ public void removeThread(long threadId) {
...
+        }
+        ScopeStackCollector scopeStackCollector = scopeManager.getCurrentScope();
+        if (scopeStackCollector == null) {
+          continue;
+        }
+        scopeStackCollector.collect(threadInfo.getStackTrace());
       }
-      scopeStackCollector.collect(threadInfo.getStackTrace());
+    } catch (Exception ex) {
+      log.warn("Exception thrown during JMX sampling:", ex);
```
Comment (jbachorik) [inline dd-java-agent/agent-mlt/src/main/java/com/datadog/mlt/sampler/JMXSampler.java:113]: What is the chance of this happening on customer deployments?
Would it be a very rare occasion only?

Comment (jpbempel) [reply dd-java-agent/agent-mlt/src/main/java/com/datadog/mlt/sampler/JMXSampler.java:113]: well in fact it's happening in staging for prof-analyzer-java 😁

Comment (jbachorik) [reply dd-java-agent/agent-mlt/src/main/java/com/datadog/mlt/sampler/JMXSampler.java:113]: In that case perhaps `INFO` and some innocent message :) Customers tend to be quite spooked when seeing WARN and ERROR in their app logs :)

Comment (jpbempel) [reply dd-java-agent/agent-mlt/src/main/java/com/datadog/mlt/sampler/JMXSampler.java:113]: noted, right now, I am trying to figure out what is happening :)

Comment (jpbempel) [reply dd-java-agent/agent-mlt/src/main/java/com/datadog/mlt/sampler/JMXSampler.java]: done

File: dd-java-agent/agent-mlt/src/main/java/com/datadog/mlt/sampler/JMXSampler.java [hunk 2]
```diff
@@ -86,27 +86,34 @@ public void removeThread(long threadId) {
...
+        }
+        ScopeStackCollector scopeStackCollector = scopeManager.getCurrentScope();
+        if (scopeStackCollector == null) {
+          continue;
+        }
+        scopeStackCollector.collect(threadInfo.getStackTrace());
       }
-      scopeStackCollector.collect(threadInfo.getStackTrace());
+    } catch (Exception ex) {
+      log.info("Exception thrown during JMX sampling:", ex);
```
Comment (tylerbenson) [inline dd-java-agent/agent-mlt/src/main/java/com/datadog/mlt/sampler/JMXSampler.java:116]: This should either be at debug only or rate limited.

Comment (jpbempel) [reply dd-java-agent/agent-mlt/src/main/java/com/datadog/mlt/sampler/JMXSampler.java]: done

**Pull request merged**